Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -152,53 +186,14 @@ public CompactSampler(int sampleSize, double lambda, Random random, boolean stor
         }
         this.random = random;
         this.lambda = lambda;
+        this.initialAcceptFraction = initialAcceptFraction;
```

Comment:
Can we do something like:
```
double p = 1 - size() / capacity();
if (Math.random() <= p) {
  // accept point
} else {
  // use time-decayed sampling as before
}
```

This seems conceptually nicer to me and it doesn't require an explicit parameter. The potential downside is that every sample will accept the first point that it sees, and will accept the first few points with high probability, but they should start to diverge before the samplers are full.

Reply:
This specific notion of p would mean that the sampler is full after capacity * ln (capacity) number of elements in expectation. We can use this formula, but add the constant boost. That preserves the intent, but also ensures that the samplers are full before capacity/initialFracton (say 0.25 would mean 4 * capacity).